### PR TITLE
Remove Join from ExpressionActions

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/ExpressionAnalyzer.cpp [hunk 1]
```diff
@@ -1091,15 +1090,8 @@ ExpressionAnalysisResult::ExpressionAnalysisResult(
...
-                chain.addStep();
-
-            query_analyzer.appendJoin(chain);
-
-            join = chain.getLastActions();
-            if (!join)
-                throw Exception("No expected JOIN", ErrorCodes::LOGICAL_ERROR);
+            before_join = chain.getLastActions();
+            join = query_analyzer.appendJoin(chain);
             chain.addStep();
```
Comment (4ertus2) [inline src/Interpreters/ExpressionAnalyzer.cpp:1095]: Do we need one more step here? It looksl like we've already added one in `appendJoin()`.

**Review by 4ertus2 — APPROVED** [review]

File: src/Interpreters/ExpressionAnalyzer.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Interpreters/ExpressionAnalyzer.cpp:1095]: I suppose that we need, cause in following code we expect the last step in chain is Expression

**Comment by alexey-milovidov** [comment]
This pull requests most likely caused "Pipeline stuck" errors. Revert?

**Comment by KochetovNicolai** [comment]
@alexey-milovidov no, most likely it is #14412
Possible fix: 14683

**Pull request merged**